src/hooks/use-handle-snapshot.ts

```diff
@@ -28,12 +28,16 @@ export const useHandleSnapshot = (storeExists = true) => {
     const resp = await saveSnapshot(docToJSON())
     if ('data' in resp) {
       execCmd(trackCommands.applyAndRemoveChanges())
-      setTimeout(() => {
-        const state = useEditorStore.getState().editorState
-        if (!state) {
-          return
-        }
-        usePouchStore.getState().saveDoc(getDocWithoutTrackContent(state))
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          const state = useEditorStore.getState().editorState
+          if (!state) {
+            return
```

Comment:
all good by maybe here we can resolve or reject too, since we are in a promise now

Reply:
should it throw an error? 
 reject(new Error('State is not available'))

Reply:
yes, looks good